Add scope and PR size guidelines to `CONTRIBUTING.md` (#23133)
ultralytics/ultralytics: CONTRIBUTING.md

5. **[Commit your changes](https://docs.github.com/en/desktop/making-changes-in-a-branch/committing-and-reviewing-changes-to-your-project-in-github-desktop):** Commit your changes with concise and descriptive commit messages. If your changes address a specific issue, include the issue number (e.g., `Fix #123: Corrected calculation error.`).
6. **[Create a pull request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request):** Submit a pull request from your branch to the `main` branch of the original Ultralytics repository. Provide a clear title and a detailed description explaining the purpose and scope of your changes.

### 📏 Contribution Scope & PR Size Guidelines

**First-time contributors**

> First-time contributors are expected to submit small, focused pull requests. Large PRs are generally not accepted from first-time contributors. Contributions should be limited to bug fixes and minor issues. Feature additions are not expected at this stage.

Reviewer: I think it would be best to make this as a recommendation/guideline. If Linus Torvalds were to open a feature request PR as a first contribution (as an extreme example), I don't think we'd disregard it b/c it was his first contribution to _this_ repo. There are likely lots of very capable developers that may be first time contributors to Ultralytics, and propose a solid feature change.

I also think that adding specificity about what a "large" change looks like would be helpful here. Everyone's context for "large" is different. I may even change the word to "significant" as it can hold several meanings, which is applicable to what we mean here. As an example, if I contribute a change that modifies the arguments of a public function, the lines of code changed could be small, yet I would still consider it a _significant_ change.

**Established contributors**
Contributors with 5 or more merged PRs may propose and implement new features. Feature PRs must follow a prior feature request issue that has been adequately discussed and explicitly greenlit by maintainers. Even for established contributors, PRs should remain reasonably scoped with a clear limit on the amount of code changed.

**PR size and review time**

> The larger the change set, the longer the review process will take. Smaller, well-scoped PRs significantly increase the likelihood of timely review and merge.

Reviewer: ```suggestion
The larger the proposed changes to the code, the longer the review process will take. Smaller, narrowly-scoped PRs can significantly increase the likelihood of a timely review and merge.
```
